### PR TITLE
Port XslTransformation to .NET Core

### LINKED ISSUE #3368: XslTransform task not available in .NET Core
### Steps to reproduce

Project file (test.msbuild)
```xml
<Project>
   <Target Name="Foo">
      <XslTransform />
   </Target>
</Project>
```

Command line
```
dotnet msbuild /t:Foo
```
### Expected  behavior
msbuild complains that not all necessary parameters are given to the task:

>  error MSB4044: The "XslTransform" task was not given a value for the required parameter "...".

(of course, if all required parameter would be given it should execute the task...)

### Environment data
`dotnet --version` output:
>2.1.201

dotnet msbuild output:
>Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core

Note:
This is similar to https://github.com/Microsoft/msbuild/issues/1731 which was fixed by https://github.com/Microsoft/msbuild/commit/0956cef84e35eb05b2756416059d8cbe00430c5f

### PR DESCRIPTION
**Pull request by BenVillalobos**
Fixes #3368 

XslTransformation was ported over to .NET Core. though any functionality having to do with compiled DLLs had to be cut. This functionality is locked behind the flag: `FEATURE_COMPILED_XSL` which is defined under non-mono builds.

### CODE REVIEW

File: src/Tasks.UnitTests/XslTransformation_Tests.cs [hunk 1]
```diff
@@ -1227,7 +1182,9 @@ private void CompileDoubleType(string outputFile)
 
             asmBldr.Save(Path.GetFileName(outputFile), PortableExecutableKinds.ILOnly, ImageFileMachine.I386);
         }
+
+#endif
         #endregion
     }
 #endif
-}
+    }
```
Comment (rainersigwald) [inline src/Tasks.UnitTests/XslTransformation_Tests.cs]: nit 

```suggestion
}
```

File: src/Tasks/XslTransformation.cs [hunk 2]
```diff
@@ -467,6 +467,8 @@ public XslCompiledTransform LoadXslt(bool useTrustedSettings)
                         Type t = FindType(assemblyPath, typeName);
                         xslct.Load(t);
                         break;
+#else
+                        throw new Exception("This is not supported on .NET Core.");
```
Comment (rainersigwald) [inline src/Tasks/XslTransformation.cs]:
```suggestion
                        throw new PlatformNotSupportedException("Precompiled XSLTs not supported on .NET Core.");
```

Comment (rainersigwald) [reply src/Tasks/XslTransformation.cs]: And can you catch this specifically in the relevant block in `Execute` and log a new MSB error?

Comment (livarcocc) [reply src/Tasks/XslTransformation.cs]: Yes. Throwing blanket Exception makes me nervous.

Comment (BenVillalobos) [reply src/Tasks/XslTransformation.cs]: Got it

File: src/Tasks/Resources/Strings.resx [hunk 3]
```diff
@@ -2263,6 +2263,10 @@
   <data name="XslTransform.UseTrustedSettings" xml:space="preserve">
     <value>The usage of the document() method and embedded scripts is prohibited by default, due to risks of foreign code execution.  If "{0}" is a trusted source that requires those constructs, please set the "UseTrustedSettings" parameter to "true" to allow their execution.</value>
   </data>
+    <data name="XslTransform.PrecompiledXsltError" xml:space="preserve">
+    <value>MSB3705: Precompiled XSLTs are not supported in .NET Core</value>
```
Comment (rainersigwald) [inline src/Tasks/Resources/Strings.resx]:
```suggestion
    <value>MSB3705: XslCompiledDllPath is not supported when building with .NET Core.</value>
```

**Pull request merged**